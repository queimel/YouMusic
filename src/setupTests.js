import requestAnimationFrame from './tempPolyfills';
 
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = require('jest-fetch-mock')
window.fetch = require('jest-fetch-mock')
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

