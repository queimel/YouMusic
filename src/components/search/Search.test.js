import React from 'react';
import { shallow } from 'enzyme';
import setupTests from '../../setupTests';
import tempPolyfills from '../../tempPolyfills';
import Search from './search';


const search = shallow(<Search />);


it('renders correctly', () => {
    expect(search).toMatchSnapshot();
});