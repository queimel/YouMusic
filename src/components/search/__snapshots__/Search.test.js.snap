// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action=""
      >
        <article
          className="input"
        >
          <input
            name="search"
            onKeyUp={undefined}
            placeholder="buscar"
            type="input"
          />
        </article>
        <button
          type="submit"
        >
          <i
            className="fas fa-sync-alt"
          />
        </button>
      </form>,
      "id": "search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "",
        "children": Array [
          <article
            className="input"
          >
            <input
              name="search"
              onKeyUp={undefined}
              placeholder="buscar"
              type="input"
            />
          </article>,
          <button
            type="submit"
          >
            <i
              className="fas fa-sync-alt"
            />
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="search"
              onKeyUp={undefined}
              placeholder="buscar"
              type="input"
            />,
            "className": "input",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "search",
              "onKeyUp": undefined,
              "placeholder": "buscar",
              "type": "input",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-sync-alt"
            />,
            "type": "submit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-sync-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action=""
        >
          <article
            className="input"
          >
            <input
              name="search"
              onKeyUp={undefined}
              placeholder="buscar"
              type="input"
            />
          </article>
          <button
            type="submit"
          >
            <i
              className="fas fa-sync-alt"
            />
          </button>
        </form>,
        "id": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "",
          "children": Array [
            <article
              className="input"
            >
              <input
                name="search"
                onKeyUp={undefined}
                placeholder="buscar"
                type="input"
              />
            </article>,
            <button
              type="submit"
            >
              <i
                className="fas fa-sync-alt"
              />
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                name="search"
                onKeyUp={undefined}
                placeholder="buscar"
                type="input"
              />,
              "className": "input",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "search",
                "onKeyUp": undefined,
                "placeholder": "buscar",
                "type": "input",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-sync-alt"
              />,
              "type": "submit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-sync-alt",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
